buildpack/util.py

```diff
@@ -122,7 +122,11 @@ def download_and_unpack(url, destination, cache_dir="/tmp/downloads"):
             cached_location=cached_location, dest=destination
         )
     )
-    if file_name.endswith(".tar.gz") or file_name.endswith(".tgz"):
```

Comment:
Be sure to check if the extensions are picked up in `_delete_other_versions` as well. If they're not, this will cause issues in application build caches on Cloud Foundry.